servicetalk-grpc-gradle-plugin/src/main/groovy/io/servicetalk/grpc/gradle/plugin/ServiceTalkGrpcPlugin.groovy

```diff
@@ -67,6 +62,49 @@ class ServiceTalkGrpcPlugin implements Plugin<Project> {
         throw new InvalidUserDataException("Please set `serviceTalkGrpc.protobufVersion`.")
       }
 
+      // If this project is outside of ServiceTalk's gradle build we need to add an explicit dependency on the
+      // uber jar which contains the protoc logic, as otherwise the grpc-gradle-plugin will only add a dependency
+      // on the executable script
+      File uberJarFile
+      String scriptNamePrefix
+      if (serviceTalkProtocPluginPath) {
+        scriptNamePrefix = "servicetalk-grpc-protoc-" + project.version
+        uberJarFile = new File(serviceTalkProtocPluginPath)
+      } else {
+        scriptNamePrefix = "servicetalk-grpc-protoc-$serviceTalkVersion"
+        def stGrpcProtocDep =
+            project.getDependencies().create("io.servicetalk:servicetalk-grpc-protoc:$serviceTalkVersion:all")
+        compileOnlyDeps.add(stGrpcProtocDep)
+        testCompileOnlyDeps.add(stGrpcProtocDep)
+
+        uberJarFile = project.configurations.compileOnly.find { it.name.startsWith("servicetalk-grpc-protoc") }
+        if (uberJarFile == null) {
+          throw new IllegalStateException("failed to find the servicetalk-grpc-protoc:$serviceTalkVersion:all")
+        }
+      }
+
+      File scriptExecutableFile
+      try {
+        if (org.gradle.internal.os.OperatingSystem.current().isWindows()) {
+          scriptExecutableFile = File.createTempFile(scriptNamePrefix, ".bat")
```

Comment:
generation of temp files is not ideal ... but a reliable way to make sure we use the jar file that is actually resolved for each build. I'm open to alternatives @idelpivnitskiy

Reply:
how grpc-java solves this issue?

Reply:
they don't have to deal with this because they write their plugin in c++ and publish a single executable for each platform.